shared/src/native-src/loader.cpp

```diff
@@ -1391,6 +1536,125 @@ namespace shared
 
     //
 
+    mdToken Loader::GetModuleEntryPointToken(LPCBYTE moduleBaseLoadAddress, DWORD moduleFlags)
+    {
+        if (moduleBaseLoadAddress == nullptr || (moduleFlags & COR_PRF_MODULE_RESOURCE) != 0 ||
+            (moduleFlags & COR_PRF_MODULE_WINDOWS_RUNTIME) != 0)
+        {
+            return mdTokenNil;
+        }
+
+        try
+        {
+
+            // Get NT Headers from module DOS Header
+            const auto pHeaders = moduleBaseLoadAddress + VAL32(((IMAGE_DOS_HEADER*) moduleBaseLoadAddress)->e_lfanew);
+            const auto ntHeaders = (IMAGE_NT_HEADERS*) (pHeaders);
+
+            // We detect the module bitness and extract the IMAGE_DATA_DIRECTORY for CLR Data
+            IMAGE_DATA_DIRECTORY directoryEntry;
+            if (ntHeaders->OptionalHeader.Magic == VAL16(IMAGE_NT_OPTIONAL_HDR32_MAGIC))
+            {
+                directoryEntry =
+                    ((IMAGE_NT_HEADERS32*) pHeaders)->OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_COMHEADER];
+            }
+            else
+            {
+                directoryEntry =
+                    ((IMAGE_NT_HEADERS64*) pHeaders)->OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_COMHEADER];
+            }
+
+            const auto clrDataVirtualAddress = VAL32(directoryEntry.VirtualAddress);
+
+            LPCBYTE pCorHeader = nullptr;
+
+            // Special handling in case the module has a Flat or BySection layout.
+            // https://github.com/dotnet/runtime/issues/11878#issuecomment-457149220
+            if ((moduleFlags & COR_PRF_MODULE_FLAT_LAYOUT) != 0)
+            {
+                Debug("Loader::GetModuleEntryPointToken: Module with flat layout detected.");
+                IMAGE_SECTION_HEADER* sectionRet = nullptr;
+                const auto pSection = pHeaders + FIELD_OFFSET(IMAGE_NT_HEADERS, OptionalHeader) +
+                                      VAL16(ntHeaders->FileHeader.SizeOfOptionalHeader);
+
+                Debug("Loader::GetModuleEntryPointToken: Number of sections: " +
```

Comment:
Don't we need to check if (_loaderOptions.LogDebugIsEnabled) before calling Debug() in that method?

Reply:
Yes you're right, done in 742ab68